src/proof/veriT/veriT_post_processor.cpp

```diff
@@ -0,0 +1,1163 @@
+    }
+    // ======== Double negation elimination
+    // Children: (P:(not (not F)))
+    // Arguments: none
+    // ---------------------
+    // Conclusion: (F)
+    //
+    // proof rule: not_not
+    // proof term: (VP:(or (not (not (not F))) F))
+    // premises: 
+    // args: ()
+    //
+    // proof rule: resolution
+    // proof term: (F)
+    // premises: (VP:(or (not (not (not F))) F)) (P:(not (not F)))
+    // args: ()
+    case PfRule::NOT_NOT_ELIM:{
+      Node vp = d_nm->mkNode(kind::OR,d_nm->mkNode(kind::NOT,children[0]),res);
+      return addVeriTStep(vp,VeriTRule::NOT_NOT,{},{},*cdp)
+             && addVeriTStep(res,VeriTRule::RESOLUTION,{children[0],vp},{},*cdp);
+    }  
+    // ======== Contradiction
+    // Children: (P1:F P2:(not F))
+    // Arguments: ()
+    // ---------------------
+    // Conclusion: false
+    //
+    // proof rule: resolution
+    // proof term: ()
+    // premises: (P1:F) (P2:(not F))
+    // args: ()
+    case PfRule::CONTRA:{
+      new_args.push_back(d_nm->mkConst<Rational>(static_cast<unsigned>(VeriTRule::RESOLUTION)));
+      new_args.push_back(Node::null()); //print this as empty	
+      cdp->addStep(res,PfRule::VERIT_RULE,children,new_args);
+      return true;	
+    }
+    // ======== And elimination
+    // Children: (P:(and F1 ... Fn))
+    // Arguments: (i)
+    // ---------------------
+    // Conclusion: (Fi)
+    //
+    // proof rule: and
+    // proof term: (Fi)
+    // premises: (P:(and F1 ... Fn))
+    // args: ()
+    case PfRule::AND_ELIM:{
+      return addVeriTStep(res,VeriTRule::AND,children,{},*cdp);
+    }
+    // ======== And introduction
+    // Children: (P1:F1 ... Pn:Fn))
+    // Arguments: ()
+    // ---------------------
+    // Conclusion: (and P1 ... Pn)
+    //
+    // First the tautology (or (and F1 ... Fn) (not F1) ... (not Fn))) is introduced. Then, the resolution rule is applied n times where Vpi+1 is obtained by applying resolution to VPi:(or (and F1 ... Fn) (not Fi) ... (not Fn))) and Fi.
+    //
+    // proof rule: and_neg
+    // proof term: (VP1:(or (and F1 ... Fn) (not F1) ... (not Fn)))
+    // premises: ()
+    // args: ()
+    //
+    // proof rule: resolution
+    // proof term: (VP2:(or (and F1 ... Fn) (not F2) ... (not Fn)))
+    // premises: (VP1:(or (and F1 ... Fn) (not F1) ... (not Fn))) P1
+    // args: ()
+    //
+    // ...
+    //
+    // proof rule: resolution
+    // proof term: (VP:(or (and F1 ... Fn))
+    // premises: (VPn:(or (and F1 ... Fn) (not Fn))) Fn
+    // args: () 
+    case PfRule::AND_INTRO:{
+      bool success = true;
+      std::vector<Node> neg_Nodes;
+      neg_Nodes.push_back(d_nm->mkNode(kind::AND,children));
+      for(int i = 0; i < children.size(); i++){
+        neg_Nodes.push_back(d_nm->mkNode(kind::NOT,children[i]));
+      }
+
+      Node and_neg_Node = d_nm->mkNode(kind::OR,neg_Nodes);	
+      success = success && addVeriTStep(and_neg_Node,VeriTRule::AND_NEG,{},{},*cdp);
```

Comment:
If you failed here you'd just return `false`, no? No need for the `success` variable.